coverage/docs/design.md

```diff
@@ -0,0 +1,138 @@
+# Overview
+This code coverage tool has two major modes of operation.
+## In post-submit workflow 
+Post-submit workflow starts when a commit to the base branch happens, e.g. when a pull request is merged
+
+Tt runs code coverage and generates the following artifacts in a GCS bucket
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+  - code coverage profile, which is produced by "go test -coverprofile" and contains all block level
+   code coverage data. The profile will be used later in presubmit workflow as a base for comparison
+
+## In pre-submit workflow
+Pre-submit workflow starts when a pull requested is created or updated with new commit
+  - it runs code coverage on target directories and compares the new result with the one stored by 
+  the post-submit workflow and generate coverage difference. It reports coverage changes to the 
+  pull request as a comment by a robot github account. 
+  - it uses go tools to generate line by line coverage and stores the result in html, 
+  with a link as part of the robot comment mentioned above.
+  - it can be configured to return with a non-zero status if coverage falls below threshold.
+
+## Users
+The pre-submit tool is intended for a developer to see the impact on code coverage of his/her commit. It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.
+
+The post-submit testgrid report is for repo managers and/or test infra team to monitor code coverage stats over time.
+
+## Programming Language Supported
+The code coverage tool only collect code coverage for Go files
+
+
+# Design of Test Coverage Tool
+We pack the test coverage feature in a container, that is triggered to run by a CI/CD system such as prow, in response to Github events such as pulls and merges.The behavior varies in presubmit and postsubmit workflows, which is discussed in latter sections. 
```

Comment:
> that is triggered to run by a CI/CD system such as prow, in response to Github events such as pulls and merges

If you want to include information like this that is not specific to the job, please include it in the "usage with Prow" section.